crates/core/tedge_mapper/src/core/mapper.rs

```diff
@@ -19,17 +20,17 @@ pub async fn create_mapper(
     info!("{} starting", app_name);
 
     let mapper_config = converter.get_mapper_config();
-    let mqtt_client = Connection::new(&mqtt_config(
-        app_name,
-        &mqtt_host,
-        mqtt_port,
-        mapper_config.in_topic_filter.clone(),
-    )?)
-    .await?;
+    let mut topics = mapper_config.in_topic_filter.clone();
+    topics.add(format!("tedge/health-check/{}", app_name).as_str())?;
```

Comment:
I am not sure which is better, to add a topic filter here, or add the filter to both `c8y` and `az` mapper when they construct MapperConfig.

https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_mapper/src/c8y/converter.rs#L82

Reply:
I initially had them in those converter modules but then moved it here. My rationale for keeping it here was that, this health check mechanism is a responsibility of the "mapper abstraction" that abstracts the MQTT responsibilities rather than the "converter abstraction" that's just about the message conversion logic. Since health check is not a "conversion", it made sense to keep it at the mapper level itself.

The additional benefit is that we'll get "free health check out of the box" for any future cloud mappers that conform to similar code structure.